nest-forms-backend/src/nases/utils-services/tokens.nases.service.ts

```diff
@@ -292,24 +315,59 @@ export default class NasesUtilsService {
     return result; */
   }
 
-  private formatXmlToOneLine(xml: string): string {
-    const result = xml
-      .replaceAll(/\s+/g, ' ') // replace all multiple whitespaces with just one space
-      .replaceAll(/>\s+</g, '><') // delete all whitespaces between tags
-      .replaceAll(/\s*<\s*/g, '<') // delete all whitespaces (spaces) around tag marks
-      .replaceAll(/\s*>\s*/g, '>')
-      .replaceAll(/\s*\/>/g, '/>')
-      .replaceAll(/<\/\s*/g, '</')
-      .trim()
+  private async handleFormConversion(
+    form: Forms,
+    isSigned: boolean,
+  ): Promise<string | object | null> {
+    let message: string | object | null = null
+    const parser = new Parser({ normalize: true, trim: true })
 
-    return result
+    if (form.formDataBase64) {
+      message = form.formDataBase64
+    }
+
+    if (!isSigned) {
+      try {
+        const messageXml = await this.convertService.convertJsonToXmlV2(
+          {
+            formId: form.id,
+            slug: form.formDefinitionSlug,
+            jsonData: form.formDataJson,
+          },
+          null,
+        )
+
+        parser.parseString(messageXml, (err, result) => {
```

Comment:
How does this work? Isn't callback executed after the function? If it works in the test it's just pure coincidence and is left to timing.

Please use `parseStringPromise` and `async/await`.